js/exporter/jspdf/export.js

```diff
@@ -0,0 +1,181 @@
+        }
+
+        return fullOptions;
+    },
+
+    getDefaultAutoTableOptions: function() {
+        return {
+            theme: 'plain',
+            tableLineColor: 149,
+            tableLineWidth: 0.1,
+            styles: {
+                textColor: 51,
+                lineColor: 149,
+                lineWidth: 0
+            },
+            columnStyles: {},
+            headStyles: {
+                fontStyle: 'normal',
+                textColor: 149,
+                lineWidth: 0.1
+            },
+            bodyStyles: {
+                lineWidth: 0.1
+            },
+            head: [],
+            body: []
+        };
+    },
+
+    export: function(options, privateOptions) {
+        const {
+            jsPDFDocument,
+            autoTableOptions,
+            component,
+            keepColumnWidths,
+            selectedRowsOnly
+        } = options;
+
+        const tableWidth = component.option('width');
+        const dataProvider = component.getDataProvider(selectedRowsOnly);
+
+        return new Promise((resolve) => {
+            dataProvider.ready().done(() => {
+                const columns = dataProvider.getColumns();
+                const styles = dataProvider.getStyles();
+                const dataRowsCount = dataProvider.getRowsCount();
+
+                this.setTableWidth(autoTableOptions, tableWidth);
+                this.setColumnWidths(autoTableOptions, columns, keepColumnWidths);
+
+                for(let rowIndex = 0; rowIndex < dataRowsCount; rowIndex++) {
+
+                    let rowType = null;
+                    const row = [];
+
+                    for(let cellIndex = 0; cellIndex < columns.length; cellIndex++) {
+
+                        const cellData = dataProvider.getCellData(rowIndex, cellIndex, true);
+                        const cellStyle = styles[dataProvider.getStyleId(rowIndex, cellIndex)];
+                        const gridCell = cellData.cellSourceData;
+                        const pdfCell = {
+                            content: cellData.value,
+                            styles: {}
+                        };
+
+                        this.assignCellStyle(pdfCell, gridCell, columns[cellIndex], cellStyle);
+
+                        rowType = rowType || gridCell.rowType;
+                        if(rowType === 'header') {
+                            if(pdfCell.content !== '') { row.push(pdfCell); }
+                        } else {
+                            row.push(pdfCell);
+                        }
+                    }
+
+                    this.addRow(autoTableOptions, rowType, row);
+                }
+
+                jsPDFDocument.autoTable(autoTableOptions);
+
+                resolve(autoTableOptions);
+            }).always(() => {
+
+            });
+        });
+    },
+
+    addRow: function(autoTableOptions, rowType, row) {
```

Comment:
Six lines of code instead of two lines.
Why do not place these 2 instead of addRow call?